accounts-journey/src/main/kotlin/com/backbase/accounts_journey/presentation/accountdetail/mapper/AccountDetailUiMapper.kt

```diff
@@ -0,0 +1,66 @@
+package com.backbase.accounts_journey.presentation.accountdetail.mapper
+
+import com.backbase.accounts_journey.common.mapCatching
+import com.backbase.accounts_journey.configuration.AccountsJourneyConfiguration
+import com.backbase.accounts_journey.configuration.icon.IconsConfiguration
+import com.backbase.accounts_journey.domain.model.AccountType
+import com.backbase.accounts_journey.domain.model.account_detail.AccountDetail
+import com.backbase.accounts_journey.presentation.accountdetail.model.AccountDetailUiModel
+import com.backbase.android.design.amount.AmountFormat
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.time.format.DateTimeFormatter
+
+/**
+ *  A AccountDetail mapper from domain models to UI model.
+ *
+ *  Created by Backbase R&D B.V on 16/11/2023.
+ */
+class AccountDetailUiMapper(accountsJourneyConfiguration: AccountsJourneyConfiguration) {
+
+    private val iconsConfiguration: IconsConfiguration by lazy {
+        accountsJourneyConfiguration.iconsConfiguration
+    }
+
+    fun mapToUi(domain: AccountDetail): AccountDetailUiModel? {
+        return domain.mapCatching {
+            AccountDetailUiModel(
+                id = this.id,
+                name = this.name!!,
```

Comment:
Prevent double bang operators. Essentially with this you state in code: crash if it does not exist

Reply:
Things that are nullable in our endpoints simply are nullable. That doesn't make the data invalid. I don't really like the solution of mapCatching. IMO it should not be null. If mapping doesn't work out, the reason should be known. Throwing an exception or having a Result object providing context makes sense.

The double bang operator to me is always better to be avoided. It leads to making code non-null safe. Now your code technically is, but the engineer/code reviewer needs to know about how it is used to understand that this double bang operator works as expected.

Having a validator for the object makes sense to me though

Reply:
I'd be very happy to talk about the value of our current UseCase layer if all it does it pass on call to client and map 1-1 to a Domain model which is not domain specific, but the current case is that the decision what is shown on the UI by the customer is done with the UIDataMappers, and for that they will need the entire object in the ViewModel